modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/utils.py

```diff
@@ -13,11 +13,202 @@
 
 """Utilities for internal use by the ``HdkOnNativeDataframe``."""
 
-
+import re
 import pandas
 import pyarrow
 
+from pandas import Timestamp
 from modin.error_message import ErrorMessage
+from modin.utils import MODIN_UNNAMED_SERIES_LABEL
+from string import ascii_uppercase, ascii_lowercase, digits
+from typing import List, Tuple, Union
+
+IDX_COL_NAME = "__index__"
+ROWID_COL_NAME = "__rowid__"
+
+# Bytes 62 and 63 are encoded with 2 characters
+_BASE_EXT = ("_A", "_B")
+_BASE_LIST = tuple(ascii_uppercase + ascii_lowercase + digits) + _BASE_EXT
+_BASE_DICT = dict((c, i) for i, c in enumerate(_BASE_LIST))
+_NON_ALPHANUM_PATTERN = re.compile("[^a-zA-Z0-9]+")
+# Number of bytes in the tailing chunk
+_TAIL_LEN = {"_0": 0, "_1": 1, "_2": 2, "_3": 3}
+_RESERVED_NAMES = (MODIN_UNNAMED_SERIES_LABEL, ROWID_COL_NAME)
+_COL_TYPES = Union[str, int, float, Timestamp, None]
+_COL_NAME_TYPE = Union[_COL_TYPES, Tuple[_COL_TYPES, ...]]
+
+
+def encode_col_name(
```

Comment:
I am a bit concerned about the performance of this. `encode/decode_col_name` functions are called pretty often as I have seen and ideally we would want this code to be written in C, maybe there's already a library that does efficient encoding/decoding to alpha-num strings? But maybe we don't need to care about this much if the python's perf is okay, so, can you please measure how these functions perform and post the result?

Reply:
@AndreyPavlenko this is quite low-level and complex code. Is it possible to make the necessary utility in the HDK itself?

Reply:
> @AndreyPavlenko this is quite low-level and complex code. Is it possible to make the necessary utility in the HDK itself?

I'll try to implement it in C and, if the performance is much better, I'll submit a PR to HDK.